include/zephyr/net/http/service.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_SERVICE_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_SERVICE_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct http_resource_desc {
+	const char *resource;
+	void *detail;
+};
+
+/**
+ * @brief Define a static HTTP resource
+ *
+ * A static HTTP resource is one that is known prior to system initialization. Other resources
+ * may be discovered upon system initialization, inserted, or removed by events originating
+ * internally or externally to the system.
+ *
+ * @note The @p resource is the URL without the associated protocol, host, or URL parameters. E.g.
+ * the resource for `http://www.foo.com/bar/baz.html#param1=value1` would be `/bar/baz.html`. Every
+ * `(service, resource)` pair should be unique.
+ *
+ * @param _name Name of the resource.
+ * @param _service Name of the associated service.
+ * @param _resource Pathname-like string identifying the resource.
+ * @param _detail Implementation-specific detail associated with the resource.
+ */
+#define HTTP_RESOURCE_DEFINE(_name, _service, _resource, _detail)                                  \
+	const STRUCT_SECTION_ITERABLE_ALTERNATE(http_resource_desc_##_service, http_resource_desc, \
+						_name) = {                                         \
+		.resource = _resource,                                                             \
+		.detail = (_detail),                                                               \
+	}
+
+struct http_service_desc {
+	const char *host;
+	uint16_t *port;
```

Comment:
Any particular reason for `port` to be a pointer? If it's not intended to be NULL in any case, perhaps just store the actual port value in the structure?

Reply:
Because it could be ephemeral. It's similar to the mDNS-SD port